<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2023-04-01" />

<title>Comparing the expected mutation rate and ovserved mutation counts</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mutation_rate</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/mutation_rate">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparing the expected mutation rate and
ovserved mutation counts</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2023-04-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mutation_rate/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230228code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230228code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230228)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229mutationratetree38fef7774b3e923dbc7e5ac57a85624c50a3820etargetblank38fef77a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/mutation_rate/tree/38fef7774b3e923dbc7e5ac57a85624c50a3820e" target="_blank">38fef77</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229mutationratetree38fef7774b3e923dbc7e5ac57a85624c50a3820etargetblank38fef77a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/mutation_rate/tree/38fef7774b3e923dbc7e5ac57a85624c50a3820e" target="_blank">38fef77</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/gnomad_exp_obs.Rmd

Unstaged changes:
    Modified:   analysis/reproducing_SC.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="data" class="section level1">
<h1>Data</h1>
<p>SNV from gnomAD v3.1 whole genome</p>
</div>
<div id="chr22" class="section level1">
<h1>Chr22</h1>
<div id="considerting-the-entire-chromosome" class="section level2">
<h2>Considerting the entire chromosome</h2>
<div
id="count-the-number-of-rare-snvsaf-0.001-do-p-1-exp-mu-conversion-of-mutation-rate-to-prob.-of-being-polymorphic."
class="section level3">
<h3>Count the number of rare SNVs(AF &lt;0.001), do p = 1-exp(-mu)
conversion of mutation rate to prob. of being polymorphic.</h3>
<p>Considering the 6 mutation types as a whole</p>
<pre><code>  CHROM      POS REF ALT ID_exp QUAL_exp FILTER_exp    MR          p
1    22 10510413   G   A      .        .        low 0.128 0.12014662
2    22 10510413   G   C      .        .        low 0.051 0.04972133
3    22 10510413   G   T      .        .        low 0.073 0.07039917
4    22 10510424   A   C      .        .        low 0.030 0.02955447
5    22 10510424   A   G      .        .        low 0.073 0.07039917
6    22 10510424   A   T      .        .        low 0.013 0.01291586
        ID_obs QUAL_obs FILTER_obs    AN AC          AF
1            .        .       PASS 25930  2 7.71307e-05
2         &lt;NA&gt;     &lt;NA&gt;       &lt;NA&gt;    NA NA          NA
3         &lt;NA&gt;     &lt;NA&gt;       &lt;NA&gt;    NA NA          NA
4         &lt;NA&gt;     &lt;NA&gt;       &lt;NA&gt;    NA NA          NA
5 rs1268976573        .       PASS 27350  1 3.65631e-05
6         &lt;NA&gt;     &lt;NA&gt;       &lt;NA&gt;    NA NA          NA</code></pre>
<pre><code>[1] &quot;The observed mutation count is 40317642&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 2238558.13213428&quot;</code></pre>
<p>Considering the 6 mutation types separately</p>
<pre><code>[1] &quot;A-C&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 2435592&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 83414.7154197101&quot;</code></pre>
<pre><code>[1] &quot;A-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 8507194&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 312909.337933403&quot;</code></pre>
<pre><code>[1] &quot;A-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 2224475&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 79874.9682219173&quot;</code></pre>
<pre><code>[1] &quot;C-A&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 3929952&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 265885.771542358&quot;</code></pre>
<pre><code>[1] &quot;C-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 3894305&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 245092.520241789&quot;</code></pre>
<pre><code>[1] &quot;C-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 19326124&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 1251380.81877508&quot;</code></pre>
<pre><code>[1] &quot;fitting: observed ~ expexted + 0&quot;</code></pre>
<p><img src="figure/gnomad_exp_obs.Rmd/unnamed-chunk-3-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre><code>
Call:
lm(formula = observed ~ expected + 0)

Residuals:
      1       2       3       4       5       6 
 933264 1087160 3448891 -902943  -67713 -368197 

Coefficients:
         Estimate Std. Error t value Pr(&gt;|t|)    
expected   16.165      1.284   12.59 5.61e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1726000 on 5 degrees of freedom
Multiple R-squared:  0.9694,    Adjusted R-squared:  0.9633 
F-statistic: 158.5 on 1 and 5 DF,  p-value: 5.614e-05</code></pre>
<div
id="all-snvs-add-up-the-total-ac-allelic-counts-of-all-svns-just-add-up-the-rates."
class="section level4">
<h4>all SNVs, Add up the total AC (allelic counts) of all SVNs, just add
up the rates.</h4>
<p>Considering the 6 mutation types as a whole</p>
<pre><code>  CHROM      POS REF ALT ID_exp QUAL_exp FILTER_exp    MR       ID_obs QUAL_obs
1    22 10510413   G   A      .        .        low 0.128            .        .
2    22 10510413   G   C      .        .        low 0.051         &lt;NA&gt;     &lt;NA&gt;
3    22 10510413   G   T      .        .        low 0.073         &lt;NA&gt;     &lt;NA&gt;
4    22 10510424   A   C      .        .        low 0.030         &lt;NA&gt;     &lt;NA&gt;
5    22 10510424   A   G      .        .        low 0.073 rs1268976573        .
6    22 10510424   A   T      .        .        low 0.013         &lt;NA&gt;     &lt;NA&gt;
  FILTER_obs    AN AC          AF
1       PASS 25930  2 7.71307e-05
2       &lt;NA&gt;    NA NA          NA
3       &lt;NA&gt;    NA NA          NA
4       &lt;NA&gt;    NA NA          NA
5       PASS 27350  1 3.65631e-05
6       &lt;NA&gt;    NA NA          NA</code></pre>
<pre><code>[1] &quot;The observed mutation count is 5294309561&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 3419901.70499986&quot;</code></pre>
<p>Considering the 6 mutation types separately</p>
<pre><code>[1] &quot;A-C&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 396146941&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 85315.2040000007&quot;</code></pre>
<pre><code>[1] &quot;A-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 1834379441&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 347197.945999998&quot;</code></pre>
<pre><code>[1] &quot;A-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 295488905&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 83753.0860000007&quot;</code></pre>
<pre><code>[1] &quot;C-A&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 406443146&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 280647.931999997&quot;</code></pre>
<pre><code>[1] &quot;C-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 506767460&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 255014.787999997&quot;</code></pre>
<pre><code>[1] &quot;C-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 1855083668&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 2367972.749&quot;</code></pre>
<pre><code>[1] &quot;fitting: observed ~ expexted + 0&quot;</code></pre>
<p><img src="figure/gnomad_exp_obs.Rmd/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre><code>
Call:
lm(formula = observed ~ expected + 0)

Residuals:
       1        2        3        4        5        6 
 1495824  1693350  5486574 -1275222  1675678  1488317 

Coefficients:
         Estimate Std. Error t value Pr(&gt;|t|)    
expected    8.700      1.193   7.295 0.000758 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 2893000 on 5 degrees of freedom
Multiple R-squared:  0.9141,    Adjusted R-squared:  0.8969 
F-statistic: 53.22 on 1 and 5 DF,  p-value: 0.0007576</code></pre>
</div>
</div>
</div>
</div>
<div id="chr21" class="section level1">
<h1>CHR21</h1>
<div id="considerting-the-entire-chromosome-1" class="section level2">
<h2>Considerting the entire chromosome</h2>
<div
id="count-the-number-of-rare-snvsaf-0.001-do-p-1-exp-mu-conversion-of-mutation-rate-to-prob.-of-being-polymorphic.-1"
class="section level3">
<h3>Count the number of rare SNVs(AF &lt;0.001), do p = 1-exp(-mu)
conversion of mutation rate to prob. of being polymorphic.</h3>
<p>Considering the 6 mutation types as a whole</p>
<pre><code>  CHROM     POS REF ALT ID_exp QUAL_exp FILTER_exp    MR          p ID_obs
1    21 5030526   G   A      .        .        low 0.174 0.15970310      .
2    21 5030526   G   C      .        .        low 0.041 0.04017087   &lt;NA&gt;
3    21 5030526   G   T      .        .        low 0.030 0.02955447   &lt;NA&gt;
4    21 5030569   A   C      .        .        low 0.020 0.01980133   &lt;NA&gt;
5    21 5030569   A   G      .        .        low 0.128 0.12014662      .
6    21 5030569   A   T      .        .        low 0.020 0.01980133   &lt;NA&gt;
  QUAL_obs FILTER_obs     AN AC          AF
1        .       PASS  90870  8 8.80379e-05
2     &lt;NA&gt;       &lt;NA&gt;     NA NA          NA
3     &lt;NA&gt;       &lt;NA&gt;     NA NA          NA
4     &lt;NA&gt;       &lt;NA&gt;     NA NA          NA
5        .       PASS 107170  1 9.33097e-06
6     &lt;NA&gt;       &lt;NA&gt;     NA NA          NA</code></pre>
<pre><code>[1] &quot;The observed mutation count is 37899765&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 2071187.92267385&quot;</code></pre>
<p>Considering the 6 mutation types separately</p>
<pre><code>[1] &quot;A-C&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 2673877&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 107255.400172301&quot;</code></pre>
<pre><code>[1] &quot;A-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 9929147&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 416065.274999948&quot;</code></pre>
<pre><code>[1] &quot;A-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 2738625&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 110738.868383197&quot;</code></pre>
<pre><code>[1] &quot;C-A&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 3784484&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 252376.689491979&quot;</code></pre>
<pre><code>[1] &quot;C-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 3207593&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 204466.462090711&quot;</code></pre>
<pre><code>[1] &quot;C-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 15566039&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 980285.227535679&quot;</code></pre>
<pre><code>[1] &quot;fitting: observed ~ expexted + 0&quot;</code></pre>
<p><img src="figure/gnomad_exp_obs.Rmd/unnamed-chunk-9-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre><code>
Call:
lm(formula = observed ~ expected + 0)

Residuals:
       1        2        3        4        5        6 
  846017   840804  2818286 -1187754  -286889  -528819 

Coefficients:
         Estimate Std. Error t value Pr(&gt;|t|)    
expected   17.091      1.328   12.87 5.04e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1493000 on 5 degrees of freedom
Multiple R-squared:  0.9707,    Adjusted R-squared:  0.9648 
F-statistic: 165.7 on 1 and 5 DF,  p-value: 5.042e-05</code></pre>
<div
id="all-snvs-add-up-the-total-ac-allelic-counts-of-all-svns-just-add-up-the-rates.-1"
class="section level4">
<h4>all SNVs, Add up the total AC (allelic counts) of all SVNs, just add
up the rates.</h4>
<p>Considering the 6 mutation types as a whole</p>
<pre><code>  CHROM     POS REF ALT ID_exp QUAL_exp FILTER_exp    MR ID_obs QUAL_obs
1    21 5030526   G   A      .        .        low 0.174      .        .
2    21 5030526   G   C      .        .        low 0.041   &lt;NA&gt;     &lt;NA&gt;
3    21 5030526   G   T      .        .        low 0.030   &lt;NA&gt;     &lt;NA&gt;
4    21 5030569   A   C      .        .        low 0.020   &lt;NA&gt;     &lt;NA&gt;
5    21 5030569   A   G      .        .        low 0.128      .        .
6    21 5030569   A   T      .        .        low 0.020   &lt;NA&gt;     &lt;NA&gt;
  FILTER_obs     AN AC          AF
1       PASS  90870  8 8.80379e-05
2       &lt;NA&gt;     NA NA          NA
3       &lt;NA&gt;     NA NA          NA
4       &lt;NA&gt;     NA NA          NA
5       PASS 107170  1 9.33097e-06
6       &lt;NA&gt;     NA NA          NA</code></pre>
<pre><code>[1] &quot;The observed mutation count is 5680597233&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 2940467.23199989&quot;</code></pre>
<p>Considering the 6 mutation types separately</p>
<pre><code>[1] &quot;A-C&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 482633742&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 109518.987000001&quot;</code></pre>
<pre><code>[1] &quot;A-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 1921651248&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 458824.330999998&quot;</code></pre>
<pre><code>[1] &quot;A-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 426185898&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 115579.750000002&quot;</code></pre>
<pre><code>[1] &quot;C-A&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 479042884&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 266858.129999998&quot;</code></pre>
<pre><code>[1] &quot;C-G&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 473464105&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 211984.628999998&quot;</code></pre>
<pre><code>[1] &quot;C-T&quot;</code></pre>
<pre><code>[1] &quot;The observed mutation count is 1897619356&quot;</code></pre>
<pre><code>[1] &quot;The expected mutation rate is 1777701.405&quot;</code></pre>
<pre><code>[1] &quot;fitting: observed ~ expexted + 0&quot;</code></pre>
<p><img src="figure/gnomad_exp_obs.Rmd/unnamed-chunk-12-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre><code>
Call:
lm(formula = observed ~ expected + 0)

Residuals:
       1        2        3        4        5        6 
 1602431  1597262  5418724 -1909463  1123701  1161164 

Coefficients:
         Estimate Std. Error t value Pr(&gt;|t|)   
expected    9.830      1.523   6.454  0.00133 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 2854000 on 5 degrees of freedom
Multiple R-squared:  0.8928,    Adjusted R-squared:  0.8714 
F-statistic: 41.66 on 1 and 5 DF,  p-value: 0.001329</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.13  knitr_1.39       magrittr_2.0.3  
 [5] workflowr_1.7.0  R6_2.5.1         rlang_1.0.5      fastmap_1.1.0   
 [9] fansi_1.0.3      highr_0.9        stringr_1.4.0    tools_4.2.0     
[13] xfun_0.30        utf8_1.2.2       cli_3.3.0        git2r_0.30.1    
[17] jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2   rprojroot_2.0.3 
[21] yaml_2.3.5       digest_0.6.29    tibble_3.1.7     lifecycle_1.0.1 
[25] crayon_1.5.1     later_1.3.0      sass_0.4.1       vctrs_0.4.1     
[29] promises_1.2.0.1 fs_1.5.2         glue_1.6.2       evaluate_0.15   
[33] rmarkdown_2.14   stringi_1.7.6    bslib_0.3.1      compiler_4.2.0  
[37] pillar_1.7.0     jsonlite_1.8.0   httpuv_1.6.5     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
