---
title: "Comparing the expected mutation rate and ovserved mutation counts"
author: "XSun"
date: "2023-04-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---



# Data

SNV from gnomAD v3.1 whole genome 

# CHR22

## Considerting the entire chromosome

### Count the number of rare SNVs(AF <0.001),  do p = 1-exp(-mu) conversion of mutation rate to prob. of being polymorphic.

```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ac.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ag.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_at.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_cg.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ct.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ca.rdata")

```

- Considering the 6 mutation types collectively

```{r echo=FALSE, message=FALSE, warning=FALSE}
print(head(merged))
print(paste0("The observed mutation count is ", sum(merged$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged$p,na.rm = T)))
```

- Considering the 6 mutation types separately

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=5}
print("A-C")
print(paste0("The observed mutation count is ", sum(merged_ac$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ac$p,na.rm = T)))

print("A-G")
print(paste0("The observed mutation count is ", sum(merged_ag$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ag$p,na.rm = T)))

print("A-T")
print(paste0("The observed mutation count is ", sum(merged_at$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_at$p,na.rm = T)))


print("C-A")
print(paste0("The observed mutation count is ", sum(merged_ca$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ca$p,na.rm = T)))

print("C-G")
print(paste0("The observed mutation count is ", sum(merged_cg$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_cg$p,na.rm = T)))


print("C-T")
print(paste0("The observed mutation count is ", sum(merged_ct$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ct$p,na.rm = T)))

observed <- c(sum(merged_at$AC,na.rm = T),sum(merged_ac$AC,na.rm = T),sum(merged_ag$AC,na.rm = T),sum(merged_ct$AC,na.rm = T),sum(merged_cg$AC,na.rm = T),sum(merged_ca$AC,na.rm = T))
expected <- c(sum(merged_at$p,na.rm = T),sum(merged_ac$p,na.rm = T),sum(merged_ag$p,na.rm = T),sum(merged_ct$p,na.rm = T),sum(merged_cg$p,na.rm = T),sum(merged_ca$p,na.rm = T))

print("fitting: observed ~ expexted + 0")
fit <- lm(observed ~ expected + 0)
plot(expected, observed)
abline(fit)
print(summary(fit))

```


### all SNVs, Add up the total AC (allelic counts) of all SVNs, just add up the rates.
```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ac.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ag.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_at.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_cg.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ct.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ca.rdata")
```


- Considering the 6 mutation types collectively

```{r echo=FALSE, message=FALSE, warning=FALSE}
print(head(merged))
print(paste0("The observed mutation count is ", sum(merged$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged$MR,na.rm = T)))
```

Considering the 6 mutation types separately

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=5}
print("A-C")
print(paste0("The observed mutation count is ", sum(merged_ac$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ac$MR,na.rm = T)))

print("A-G")
print(paste0("The observed mutation count is ", sum(merged_ag$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ag$MR,na.rm = T)))

print("A-T")
print(paste0("The observed mutation count is ", sum(merged_at$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_at$MR,na.rm = T)))


print("C-A")
print(paste0("The observed mutation count is ", sum(merged_ca$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ca$MR,na.rm = T)))

print("C-G")
print(paste0("The observed mutation count is ", sum(merged_cg$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_cg$MR,na.rm = T)))


print("C-T")
print(paste0("The observed mutation count is ", sum(merged_ct$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ct$MR,na.rm = T)))

obseved <- c(sum(merged_at$AC,na.rm = T),sum(merged_ac$AC,na.rm = T),sum(merged_ag$AC,na.rm = T),sum(merged_ct$AC,na.rm = T),sum(merged_cg$AC,na.rm = T),sum(merged_ca$AC,na.rm = T))
expected <- c(sum(merged_at$MR,na.rm = T),sum(merged_ac$MR,na.rm = T),sum(merged_ag$MR,na.rm = T),sum(merged_ct$MR,na.rm = T),sum(merged_cg$MR,na.rm = T),sum(merged_ca$MR,na.rm = T))

print("fitting: observed ~ expexted + 0")
fit <- lm(observed ~ expected + 0)
plot(expected, observed)
abline(fit)
print(summary(fit))
```


# CHR21


## Considerting the entire chromosome

### Count the number of rare SNVs(AF <0.001),  do p = 1-exp(-mu) conversion of mutation rate to prob. of being polymorphic.

```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ac_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ag_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_at_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_cg_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ct_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_rare_ca_chr21.rdata")

```

Considering the 6 mutation types collectively

```{r echo=FALSE, message=FALSE, warning=FALSE}
print(head(merged))
print(paste0("The observed mutation count is ", sum(merged$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged$p,na.rm = T)))
```

Considering the 6 mutation types separately

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=5}
print("A-C")
print(paste0("The observed mutation count is ", sum(merged_ac$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ac$p,na.rm = T)))

print("A-G")
print(paste0("The observed mutation count is ", sum(merged_ag$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ag$p,na.rm = T)))

print("A-T")
print(paste0("The observed mutation count is ", sum(merged_at$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_at$p,na.rm = T)))


print("C-A")
print(paste0("The observed mutation count is ", sum(merged_ca$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ca$p,na.rm = T)))

print("C-G")
print(paste0("The observed mutation count is ", sum(merged_cg$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_cg$p,na.rm = T)))


print("C-T")
print(paste0("The observed mutation count is ", sum(merged_ct$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ct$p,na.rm = T)))

observed <- c(sum(merged_at$AC,na.rm = T),sum(merged_ac$AC,na.rm = T),sum(merged_ag$AC,na.rm = T),sum(merged_ct$AC,na.rm = T),sum(merged_cg$AC,na.rm = T),sum(merged_ca$AC,na.rm = T))
expected <- c(sum(merged_at$p,na.rm = T),sum(merged_ac$p,na.rm = T),sum(merged_ag$p,na.rm = T),sum(merged_ct$p,na.rm = T),sum(merged_cg$p,na.rm = T),sum(merged_ca$p,na.rm = T))

print("fitting: observed ~ expexted + 0")
fit <- lm(observed ~ expected + 0)
plot(expected, observed)
abline(fit)
print(summary(fit))

```


### all SNVs, Add up the total AC (allelic counts) of all SVNs, just add up the rates.
```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ac_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ag_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_at_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_cg_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ct_chr21.rdata")
load("/project/xinhe/xsun/mutation_rate/1.exp_obs_gnomad/results/obs_exp_merged_all_ca_chr21.rdata")
```


Considering the 6 mutation types collectively

```{r echo=FALSE, message=FALSE, warning=FALSE}
print(head(merged))
print(paste0("The observed mutation count is ", sum(merged$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged$MR,na.rm = T)))
```

Considering the 6 mutation types separately

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=5}
print("A-C")
print(paste0("The observed mutation count is ", sum(merged_ac$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ac$MR,na.rm = T)))

print("A-G")
print(paste0("The observed mutation count is ", sum(merged_ag$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ag$MR,na.rm = T)))

print("A-T")
print(paste0("The observed mutation count is ", sum(merged_at$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_at$MR,na.rm = T)))


print("C-A")
print(paste0("The observed mutation count is ", sum(merged_ca$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ca$MR,na.rm = T)))

print("C-G")
print(paste0("The observed mutation count is ", sum(merged_cg$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_cg$MR,na.rm = T)))


print("C-T")
print(paste0("The observed mutation count is ", sum(merged_ct$AC,na.rm = T)))
print(paste0("The expected mutation rate is ", sum(merged_ct$MR,na.rm = T)))

obseved <- c(sum(merged_at$AC,na.rm = T),sum(merged_ac$AC,na.rm = T),sum(merged_ag$AC,na.rm = T),sum(merged_ct$AC,na.rm = T),sum(merged_cg$AC,na.rm = T),sum(merged_ca$AC,na.rm = T))
expected <- c(sum(merged_at$MR,na.rm = T),sum(merged_ac$MR,na.rm = T),sum(merged_ag$MR,na.rm = T),sum(merged_ct$MR,na.rm = T),sum(merged_cg$MR,na.rm = T),sum(merged_ca$MR,na.rm = T))

print("fitting: observed ~ expexted + 0")
fit <- lm(observed ~ expected + 0)
plot(expected, observed)
abline(fit)
print(summary(fit))
```